openslides_backend/action/actions/poll/stop.py

```diff
@@ -36,8 +38,98 @@ def update_instance(self, instance: Dict[str, Any]) -> Dict[str, Any]:
             [
                 "poll_couple_countdown",
                 "poll_countdown_id",
+                "users_enable_vote_weight",
             ],
         )
         if meeting.get("poll_couple_countdown") and meeting.get("poll_countdown_id"):
             self.control_countdown(meeting["poll_countdown_id"], "reset")
+
+        # calculate votescast, votesvalid, votesinvalid
+        voted_ids = poll.get("voted_ids", [])
+        instance["votescast"] = f"{len(voted_ids)}.000000"
+        if not meeting.get("users_enable_vote_weight") or not voted_ids:
+            instance["votesvalid"] = instance["votescast"]
+        else:
+            gmr = GetManyRequest(
+                Collection("user"), voted_ids, [f"vote_weight_${poll['meeting_id']}"]
+            )
+            gm_result = self.datastore.get_many([gmr])
+            users = gm_result.get(Collection("user"), {}).values()
+            instance["votesvalid"] = str(
+                sum(
+                    [
+                        Decimal(
+                            entry.get(f"vote_weight_${poll['meeting_id']}", "1.000000")
+                        )
+                        for entry in users
+                    ]
+                )
+            )
+        instance["votesinvalid"] = "0.000000"
+
+        # set entitled users at stop.
+        instance["entitled_users_at_stop"] = self.get_entitled_user_ids(poll)
         return instance
+
+    def get_entitled_user_ids(self, poll: Dict[str, Any]) -> List[Any]:
+        entitled_users = []
+        entitled_users_ids = set()
+        all_voted_users = poll.get("voted_ids", [])
+
+        # get all users from the groups.
+        meeting = self.datastore.get(
+            FullQualifiedId(Collection("meeting"), poll["meeting_id"]), ["group_ids"]
+        )
+        gmr = GetManyRequest(
+            Collection("group"), meeting.get("group_ids", []), ["user_ids"]
+        )
+        gm_result = self.datastore.get_many([gmr])
+        groups = gm_result.get(Collection("group"), {}).values()
+
+        for group in groups:
+            user_ids = group.get("user_ids", [])
+            if not user_ids:
+                continue
+            gmr2 = GetManyRequest(
+                Collection("user"),
+                list(user_ids),
+                [
+                    "id",
+                    "is_present_in_meeting_ids",
+                    f"vote_delegated_${poll['meeting_id']}_to_id",
```

Comment:
Maybe you should save `poll["meeting_id"]` to a local variable `meeting_id` for easier access

Reply:
I use  a local var for `meeting_id`.